analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+  },
+      
+  ## Define index date
+  index_date = campaign_start,
+
+  population = patients.satisfying(
+    """
+    age >= 18 AND age < 110
+    AND NOT has_died
+    AND covid_test_positive
+    AND eligible_binary
+    """,
+  ),
+        
+  ## COVID TEST POSITIVE
+  ## First positive SARS-CoV-2 test. [patients are eligible for treatment if diagnosed <=5d ago. Restricted to first positive covid test after index date]
+  covid_test_positive = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "binary_flag", on_or_after = "index_date - 5 days", find_first_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True,  return_expectations = {"incidence": 0.9 },
+  ),
+  covid_test_positive_date = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "date", date_format = "YYYY-MM-DD", on_or_after = "index_date - 5 days", find_first_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True, return_expectations = {"date": {"earliest": "2021-12-11", "latest": "today"}, "incidence": 0.9},
+  ),
+  covid_positive_test_type = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "case_category", on_or_after = "index_date - 5 days", find_first_match_in_period = True, 
+    restrict_to_earliest_specimen_date = True, return_expectations = {"category": {"ratios": {"LFT_Only": 0.4, "PCR_Only": 0.4, "LFT_WithPCR": 0.2}},"incidence": 0.2, },
+  ),
+  ## Second positive SARS-CoV-2 test
+  covid_test_positive_date2 = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "date", date_format = "YYYY-MM-DD", on_or_after = "covid_test_positive_date + 30 days", find_first_match_in_period = True,
+    restrict_to_earliest_specimen_date = True, return_expectations = {"date": {"earliest": "2021-12-20", "latest": "today"}, "incidence": 0.1 },
+  ),
+  ## Positive covid test 30 days prior to positive test [will only apply to patients who first tested positive towards the beginning of the study period]
+  covid_positive_previous_30_days = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", returning = "binary_flag",  between = ["covid_test_positive_date - 31 days", "covid_test_positive_date - 1 day"],
+    find_last_match_in_period = True, restrict_to_earliest_specimen_date = False, return_expectations = {"incidence": 0.05 },
+  ),
+  # positive test history
+  covid_test_positive_pre_date = patients.with_test_result_in_sgss(
+    pathogen = "SARS-CoV-2", test_result = "positive", find_first_match_in_period = True, on_or_before = "covid_test_positive_date - 30 days",
+    restrict_to_earliest_specimen_date = False, returning = "date",  date_format = "YYYY-MM-DD", return_expectations = {"date": {"earliest": "2021-12-20", "latest": "index_date"}, "incidence": 0.1},
+  ), 
+  ## Onset of symptoms of COVID-19
```

Comment:
As above, might run into problems if restricting to first onset (note this variable is not always complete so not hugely unreliable)

Reply:
I actually do not use this in the analysis, so could remove